### ISSUE TITLE
Posts containing just link get rejected

### ISSUE THREAD

**Issue by attis** [NodeBB/NodeBB#5782]
Hi,

- **NodeBB version:** 1.5.1

- **Exact steps to cause this issue:**
	1. Configure `Minimum Post Length` to `0`
	2. Try to add a post with just a link. Example:
```
<http://my.opera.com/ODIN/blog/300-million-users-and-move-to-webkit>
```
- **What you expected:**
I expect the post to be added and contain just the link.

- **What happened instead:**
Post gets rejected with `content-too-short` error (and a funny message: "Please enter a longer post. Posts should contain at least 0 character(s)."

This is due to:
```
// Trim and remove HTML (latter for composers that send in HTML, like redactor)
if (typeof item === 'string') {
	item = S(item).stripTags().s.trim();
}

if (!item || item.length < parseInt(min, 10)) { ... }
``` 
in `src/topics/create.js`.

Specifically `!item` causes the condition to apply for an empty string (which is a result of `stripTags()` execution.

NodeBB should either recognize `Minimum Post Length` set to `0` as a special case and bail out from any further checks, or removing HTML tags should be smarter and should not cut out valid Markdown links. 

Alternatively `!item` should be changed to `typeof item !== 'string'` - which sounds the easiest and should do the job too.

**Comment by julianlam** [comment]
Just wondering -- why do you wrap your urls with `<` and `>`? It may be considered an "html tag" and then fail the check as it gets sanitized away...

**Comment by attis** [comment]
Yeah... I guess my case is a little uncommon. I am migrating an existing forum that has posts in Markdown too and uses angle brackets for inline links. I know that nodebb can automatically parse out plaintext links (so without any brackets at all - and it works) but I don't want to rewrite past posts if I don't have to. 

http://spec.commonmark.org/0.27/#autolink

NodeBB composer recognizes this syntax and renders it correctly as a link but the forum engine refuses to save it.

**Comment by julianlam** [comment]
Understood, will review your analysis and decide what to do 😄

**Comment by attis** [comment]
Actually my original proposal #3 is flawed. I ended up working around by changing `!item` to `item === undefined || item === null`.

Though that's something I will be reverting once the migration completes for two reasons:
- I can't imagine anyone wanting to have 0 length posts. I turned it off specifically because the original forum does not have any restrictions and didn't want to deal with it, but once I finish the import I'll be bumping it back up. This means that my current workaround won't be needed anymore.
- Users most likely will just paste a plain text link. They won't wrap it in <>.

So the only real alternative would be to either count length of text after MD->HTML rendering or somehow recognize this MD syntax and make `stripTags()` preserve it.
